File: Java/Kode/Ponsel.java
// Class turunan dari ProdukElektronik, khusus ponsel
public class Ponsel extends ProdukElektronik {
    // Atribut private
    private double ukuranLayar;
    private int kapasitasPenyimpanan;
    private int ram;

    // Konstruktor default
    public Ponsel() {
        super();
    }

    // Konstruktor dengan parameter
    public Ponsel(String nama, String merek, int harga, int stok, int konsumsiDaya, int voltase, int garansi,
                  double ukuranLayar, int kapasitasPenyimpanan, int ram) {
        // Memanggil konstruktor induk
        super(nama, merek, harga, stok, konsumsiDaya, voltase, garansi);
        this.ukuranLayar = ukuranLayar;
        this.kapasitasPenyimpanan = kapasitasPenyimpanan;
        this.ram = ram;
    }

    // Setter dan Getter untuk setiap atribut
    public void setUkuranLayar(double ukuranLayar) {
        this.ukuranLayar = ukuranLayar;
    }
    public double getUkuranLayar() {
        return this.ukuranLayar;
    }

    public void setKapasitasPenyimpanan(int kapasitasPenyimpanan) {
        this.kapasitasPenyimpanan = kapasitasPenyimpanan;
    }
    public int getKapasitasPenyimpanan() {
        return this.kapasitasPenyimpanan;
    }

    public void setRam(int ram) {
        this.ram = ram;
    }
    public int getRam() {
        return this.ram;
    }
}

File: Python/Kode/Ponsel.py
from ProdukElektronik import ProdukElektronik

# Class turunan dari ProdukElektronik, khusus ponsel
class Ponsel(ProdukElektronik):
    # Konstruktor
    def __init__(self, nama="", merek="", harga=0, stok=0, konsumsi_daya=0, voltase=0, garansi=0, ukuran_layar=0.0, kapasitas_penyimpanan=0, ram=0):
        # Memanggil konstruktor induk
        super().__init__(nama, merek, harga, stok, konsumsi_daya, voltase, garansi)
        self.ukuran_layar = ukuran_layar
        self.kapasitas_penyimpanan = kapasitas_penyimpanan
        self.ram = ram

    # Setter dan Getter untuk setiap atribut baru
    def set_ukuran_layar(self, ukuran_layar):
        self.ukuran_layar = ukuran_layar

    def get_ukuran_layar(self):
        return self.ukuran_layar
    
    def set_kapasitas_penyimpanan(self, kapasitas_penyimpanan):
        self.kapasitas_penyimpanan = kapasitas_penyimpanan

    def get_kapasitas_penyimpanan(self):
        return self.kapasitas_penyimpanan

    def set_ram(self, ram):
        self.ram = ram

    def get_ram(self):
        return self.ram